capella2polarion/connectors/polarion_worker.py

```diff
@@ -408,14 +426,29 @@ def get_missing_link_ids(
 
     @staticmethod
     def _get_link_id(link: polarion_api.WorkItemLink) -> str:
-        return "/".join(
-            (
-                link.primary_work_item_id,
-                link.role,
-                link.secondary_work_item_project,
-                link.secondary_work_item_id,
-            )
-        )
+        secondary_id = link.secondary_work_item_id
+        if link.secondary_work_item_project:
+            secondary_id = f"{link.secondary_work_item_project}/{secondary_id}"
+        return "/".join((link.primary_work_item_id, link.role, secondary_id))
+
+    @staticmethod
+    def _filter_all_link_ids(
+        links: cabc.Iterable[polarion_api.WorkItemLink],
+    ) -> cabc.Iterator[polarion_api.WorkItemLink]:
+        for link in links:
+            if link.primary_work_item_id is None:
+                logger.error(  # type: ignore[unreachable]
+                    "Found Work Item Link without source ID: %r", link
+                )
+                continue
+
+            if link.secondary_work_item_id is None:
+                logger.error(  # type: ignore[unreachable]
+                    "Found Work Item Link without target ID: %r", link
+                )
+                continue
+
+            yield link
```

Comment:
Neither `primary_work_item_id` nor `secondary_work_item_id` are optional by their annotation - that's why the error logs are unreachable

Reply:
Yes. Because when we create Links we always pass these values. I don't see the value to add these checks anymore. It is just code for edge cases that may never happen. We only land here if the links received from the server are lacking this info.